packages/google_identity_services_web/lib/src/js_interop/google_accounts_id.dart

```diff
@@ -132,12 +166,19 @@ extension GoogleAccountsIdExtension on GoogleAccountsId {
   /// The optional [callback] is a function that gets called to report on the
   /// success of the revocation call.
   ///
-  /// The [callback] parameter must be manually wrapped in [allowInterop]
-  /// before being passed to the [revoke] function.
-  ///
   /// Method: google.accounts.id.revoke
   /// https://developers.google.com/identity/gsi/web/reference/js-reference#google.accounts.id.revoke
-  external void revoke(String hint, [RevocationResponseHandlerFn callback]);
+  void revoke(String hint, [RevocationResponseHandlerFn? callback]) {
+    if (callback == null) {
+      return _revoke(hint.toJS);
+    }
+    return _revokeWithCallback(hint.toJS, callback.toJS);
+  }
+
+  @JS('revoke')
+  external void _revoke(JSString hint);
```

Comment:
FYI - you don't need `JSString`, you can use `String` here and elsewhere if you want and we'll do the conversion for you. Of course, if you prefer it or want to minimize round-trip conversions (e.g. some other function returns a `JSString` that you pass here), feel free to keep it as is. Same with other primitives like `bool`.

Reply:
I find it confusing that some things promoting automatically vs others not, so I ended up migrating everything, ROBOT STYLE, to assume that nothing promotes.

When everything promotes, I'll come back and remove half the code :P